source/apiVolontaria/volunteer/views.py

```diff
@@ -309,6 +312,55 @@ def get_queryset(self):
             queryset = queryset.filter(user__username=username)
         return queryset
 
+    def post(self, request, *args, **kwargs):
+        response = self.create(request, *args, **kwargs)
+
+        participation = models.Participation.objects.get(
+            user=self.request.user,
+            event__id=request.data["event_id"],
+        )
```

Comment:
This DB call should be inside the if statement, no? If the creation failed, we don't need to get the participation.